extended-tracer/src/main/java/io/opentelemetry/contrib/tracer/Tracing.java

```diff
@@ -0,0 +1,374 @@
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Utility class to simplify tracing.
+ *
+ * <p>The <a
+ * href="https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/extended-tracer/README.md">README</a>
+ * explains the use cases in more detail.
+ */
+public final class Tracing {
+
+  private static final TextMapGetter<Map<String, String>> TEXT_MAP_GETTER =
+      new TextMapGetter<Map<String, String>>() {
+        @Override
+        public Set<String> keys(Map<String, String> carrier) {
+          return carrier.keySet();
+        }
+
+        @Override
+        @Nullable
+        public String get(@Nullable Map<String, String> carrier, String key) {
+          //noinspection ConstantConditions
+          return carrier == null ? null : carrier.get(key);
+        }
+      };
+  private final OpenTelemetry openTelemetry;
+
+  private final Tracer tracer;
+
+  /**
+   * Creates a new instance of {@link Tracing}.
+   *
+   * @param openTelemetry the {@link OpenTelemetry} instance
+   * @param tracerName the name of the tracer to use
+   */
+  public Tracing(OpenTelemetry openTelemetry, String tracerName) {
+    this(openTelemetry, openTelemetry.getTracer(tracerName));
+  }
+
+  /**
+   * Creates a new instance of {@link Tracing}.
+   *
+   * @param openTelemetry the {@link OpenTelemetry} instance
+   * @param tracer the {@link Tracer} to use
+   */
+  public Tracing(OpenTelemetry openTelemetry, Tracer tracer) {
+    this.openTelemetry = openTelemetry;
+    this.tracer = tracer;
+  }
+
+  /**
+   * Creates a new {@link SpanBuilder} with the given span name.
+   *
+   * @param spanName the name of the span
+   * @return the {@link SpanBuilder}
+   */
+  public SpanBuilder spanBuilder(String spanName) {
+    return tracer.spanBuilder(spanName);
+  }
+
+  /**
+   * Runs the given {@link Runnable} in a new span with the given name and with kind INTERNAL.
+   *
+   * <p>If an exception is thrown by the {@link Runnable}, the span will be marked as error, and the
+   * exception will be recorded.
+   *
+   * @param spanName the name of the span
+   * @param runnable the {@link Runnable} to run
+   */
+  public void run(String spanName, Runnable runnable) {
+    run(tracer.spanBuilder(spanName), runnable);
+  }
+
+  /**
+   * Runs the given {@link Runnable} inside of the span created by the given {@link SpanBuilder}.
+   * The span will be ended at the end of the {@link Runnable}.
+   *
+   * <p>If an exception is thrown by the {@link Runnable}, the span will be marked as error, and the
+   * exception will be recorded.
+   *
+   * @param spanBuilder the {@link SpanBuilder} to use
+   * @param runnable the {@link Runnable} to run
+   */
+  @SuppressWarnings("NullAway")
+  public void run(SpanBuilder spanBuilder, Runnable runnable) {
```

Comment:
The run variants aren't symmetric to the call variants - there aren't any overloads for handling exceptions. I think you probably did this because `Runnable#run` doesn't throw, but you added a custom functional interface in SpanCallback which combines characteristics of both Supplier and Callable, allowing the implementation's throwable to be a parameterized type. Why not do the same for Runnable for symmetry?

Reply:
great idea :+1: